crates/router/src/utils/db_utils.rs

```diff
@@ -8,3 +8,29 @@ pub fn generate_hscan_pattern_for_refund(sk: &str) -> String {
         .collect::<Vec<&str>>()
         .join("_")
 }
+
+#[cfg(feature = "kv_store")]
+pub(crate) async fn push_to_drainer_stream<T>(
+    store: &crate::services::Store,
+    redis_entry: storage_models::kv::TypedSql,
+    partition_key: crate::utils::storage_partitioning::PartitionKey<'_>,
+) -> crate::core::errors::CustomResult<(), crate::core::errors::StorageError>
+where
+    T: crate::utils::storage_partitioning::KvStorePartition,
+{
```

Comment:
Could we have this as an implementation on `Store`?

Reply:
Makes sense will do